### PR TITLE
Add logo support for commands

### LINKED ISSUE #115: Logo command doesn't work if a custom default command is set
For example:

```php
$app = new \Ahc\Cli\Application('Test App', '0.0.1');
$app->logo('My ASCII Art');
$app->command('cmd');
$app->defaultCommand('cmd');
```

```sh
# will not show the logo currently
./bin/my-app --help
```

Fixed by #112

### PR DESCRIPTION
**Pull request by kodie**
Just like Application's logo function, Commands can have them now as well in the exact same way.

```php
$app = new \Ahc\Cli\Application('Test App', '0.0.1');
$app->command('cmd')->logo('My ASCII Art');
```

Fixes #115

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -181,7 +181,8 @@ class InitCommand extends Ahc\Cli\Input\Command
                 '<bold>  init</end> <comment>--apple applet --ball ballon <arggg></end> ## details 1<eol/>' .
                 // $0 will be interpolated to actual command name
                 '<bold>  $0</end> <comment>-a applet -b ballon <arggg> [arg2]</end> ## details 2<eol/>'
-            );
+            )
+            ->logo('Ascii art logo of your app');
```
Comment (adhocore) [inline README.md:185]: logo of command?

File: src/Input/Command.php [hunk 2]
```diff
@@ -52,6 +52,8 @@ class Command extends Parser implements Groupable
 
     protected ?string $_alias = null;
 
+    protected string $logo = '';
```
Comment (adhocore) [inline src/Input/Command.php:55]: should be _logo as the command will have magic getter for user input args that may collide

**Comment by adhocore** [comment]
thanks for the pull. should we also create corresponding issues for all the pulls?

File: README.md (hunk 1 above)
Comment (kodie) [reply README.md]: Updated!

File: src/Input/Command.php (hunk 2 above)
Comment (kodie) [reply src/Input/Command.php]: Done!

File: src/Input/Command.php [hunk 3]
```diff
@@ -298,6 +318,10 @@ public function showHelp(): mixed
         $io     = $this->io();
         $helper = new OutputHelper($io->writer());
 
+        if ($this->logo) {
```
Comment (adhocore) [inline src/Input/Command.php:321]: isnt it _logo

Comment (kodie) [reply src/Input/Command.php]: Fixed!

**Review by adhocore — APPROVED** [review]

**Comment by dimtrovich** [comment]
is it necessary to define the logo when executing a command? personally, I think the current implementation is sufficient. leaving the logo when calling the default command seems more normal to me.

**Comment by kodie** [comment]
> is it necessary to define the logo when executing a command? personally, I think the current implementation is sufficient. leaving the logo when calling the default command seems more normal to me.

@dimtrovich As I mention in #115, currently, the logo doesn't show up for default commands, only for the built-in `__default__` function. So if you set a logo and then use a default command, your logo would not be displayed. Also I like the freedom of being able to use a logo (possibly even unique logo) in any of the help screens (like maybe a different version of the ASCII art in each command if I so wished).

**Comment by adhocore** [comment]
yea not super necessary but ok to have i guess as long as it doesn't mess up current working behavior or get on the way

btw pull needs rebase again :D

**Comment by adhocore** [comment]
@kodie i think instead of fixing this conflict branch, it is easier to branch off new from the latest main and [cherry-pick](https://git-scm.com/docs/git-cherry-pick) or hand-pick changes on top

**Comment by kodie** [comment]
@adhocore I'll do that now and submit a new PR.

**Pull request closed without merging**